src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java

```diff
@@ -69,7 +69,7 @@
             Run<?,?> run = get(Run.class);
             EnvironmentAction a = run == null ? null : run.getAction(EnvironmentAction.class);
             EnvVars customEnvironment = a != null ? a.getEnvironment() : run.getEnvironment(getExecution().getOwner().getListener());
-            return key.cast(EnvironmentExpander.getEffectiveEnvironment(customEnvironment, (EnvVars) value, get(EnvironmentExpander.class)));
+            return key.cast(EnvironmentExpander.getEffectiveEnvironment(customEnvironment, (EnvVars) value, get(EnvironmentExpander.class), this, get(TaskListener.class)));
```

Comment:
Appears to be causing test failures downstream in jenkinsci/workflow-cps-plugin#229 based on my local testing, see https://github.com/jenkinsci/workflow-cps-plugin/pull/229#pullrequestreview-193324238 for details.

Reply:
Due to [this code](https://github.com/jenkinsci/workflow-cps-plugin/blob/a18ed9455cb9d83afa9d715448ec5e37f0802a1d/src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java#L237-L242), which I think ought to be fixed anyway: [JENKINS-47101](https://issues.jenkins-ci.org/browse/JENKINS-47101).

In the meantime, you could change this to directly call `getListener` but pass a new `boolean` param to let that return `TaskListener.NULL` in case `!node.isActive()`.

Or simply have `getListener` fall back to `getExecution().getOwner().getListener()` when called on an inactive node, so that any messages would be sent to the overall build log. This would probably be useful anyway in terms of solving some corner cases and race conditions.

Reply:
@jglick Thanks!
I assume the same listener can be used the line above?